homeassistant/components/aussie_broadband/sensor.py

```diff
@@ -0,0 +1,193 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL, DATA_KILOBYTES, DATA_MEGABYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN, SERVICE_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+):
+    """Set up the Aussie Broadband sensor platform from a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id]["client"]
+    services = hass.data[DOMAIN][entry.entry_id]["services"]
+    update_interval = timedelta(
+        minutes=entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    )
+
+    # Create an appropriate refresh function
+    def update_data_factory(service_id):
+        async def async_update_data():
+            return await client.get_usage(service_id)
+
+        return async_update_data
+
+    entities = []
+    for service in services:
+        # Initiate a Data Update Coordinator for this endpoint
+        coordinator = DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            name=service["service_id"],
+            update_interval=update_interval,
+            update_method=update_data_factory(service[SERVICE_ID]),
+        )  # type: DataUpdateCoordinator
+        await coordinator.async_config_entry_first_refresh()
+
+        # Create the appropriate entities based on the service type
+        if service["type"] == "PhoneMobile":
+            entities.extend(
+                [
+                    AussieBroadandPhoneInternet(coordinator, service),
+                    AussieBroadandPhoneNational(coordinator, service),
+                    AussieBroadandPhoneMobile(coordinator, service),
+                    AussieBroadandPhoneSMS(coordinator, service),
+                    AussieBroadandBillingCycleLength(coordinator, service),
+                    AussieBroadandBillingCycleRemaining(coordinator, service),
+                ]
+            )
+        else:
+            entities.extend(
+                [
+                    AussieBroadandTotalUsage(coordinator, service),
+                    AussieBroadandDownloaded(coordinator, service),
+                    AussieBroadandUploaded(coordinator, service),
+                    AussieBroadandBillingCycleLength(coordinator, service),
+                    AussieBroadandBillingCycleRemaining(coordinator, service),
+                ]
+            )
+
+    async_add_entities(entities)
+    return True
+
+
+class AussieBroadandSensorEntity(CoordinatorEntity, SensorEntity):
+    """Base class for Aussie Broadband metric sensors."""
+
+    _attribute: str
+    _name: str
```

Comment:
Implement `SensorEntityDescription` to define these kind of values

Reply:
Both of these private properties are parts of attributes like name and unique id, but are not themselves members of SensorEntityDescription. Do you mean I should replace these?
`self._attr_unique_id = f"{service[SERVICE_ID]}:{self._attribute}"
self._attr_name = f"{service['name']} {self._name}"`

Reply:
`*EntityDescription`  is our new way of defining values that goes into entities, if you use that here, you can probably get away with just 1 sensor class.
Here are some recent PR's that have implemented this <https://github.com/home-assistant/core/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Amerged+Use+EntityDescription>

Reply:
So I am having issues using EntityDescription:

- The sensor name includes values from the sensor object, so does this mean I need a create a unique EntityDescription for every single entity, or can I still override this using _attr_name?
- The state values for many of the sensor are slightly different, so if I don't have unique classes for each ill have to add if then statements to make sure I am reading the right value. Some are just a key, but others are two keys deep.

Ill review the other implementations and do my best.

Reply:
You can still override if/when needed with `_attr_*`

Reply:
This has now been implemented.